### PR TITLE
Initialize default view earlier in the init process.

### PR DESCRIPTION
**Pull request by Vlatombe**
Perform the default view initialization next to loading the configuration from disk.

This is more consistent as the primary view and views are cleared just before loading from disk, which leads to a temporary inconsistent state while Jenkins initializes.

See [JENKINS-XXXXX](https://issues.jenkins.io/browse/JENKINS-XXXXX).

### Proposed changelog entries

- Developer: Initialize default view slightly earlier in the initialization process.

### Proposed upgrade guidelines

N/A

```[tasklist]
### Submitter checklist
- [X] The Jira issue, if it exists, is well-described.
- [X] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)). Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.
- [x] There is automated testing or an explanation as to why this change has no tests.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadocs, as appropriate.
- [x] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")`, if applicable.
- [x] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates, there are links to external changelogs and, if possible, full differentials.
- [ ] For new APIs and extension points, there is a link to at least one consumer.
```

### Desired reviewers

@mention

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8413"></a>

### CODE REVIEW

File: core/src/main/java/jenkins/model/Jenkins.java
Comment (jglick) [inline core/src/main/java/jenkins/model/Jenkins.java:3342]:
```suggestion
            if (views.isEmpty() || primaryView == null) {
```
while we are here

**Review by jglick — COMMENTED** [review]
Seems like a more logical spot to do this.

**Review by jglick — COMMENTED** [review]
Looks right; is there any way to test this? Like some plugin initializer that runs before `JOB_CONFIG_ADAPTED` and expects there to be a view?

**Review by jglick — APPROVED** [review]

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: test/src/test/java/jenkins/model/JenkinsTest.java
Comment (jglick) [inline test/src/test/java/jenkins/model/JenkinsTest.java:849]: By the way I think this has no actual effect—this `@Initializer` will be run regardless of which test is running (all of `jenkins-test`).

**Pull request merged**
